arviz/data/io_pymc3.py

```diff
@@ -153,18 +155,25 @@ def _extract_log_likelihood(self):
         """Compute log likelihood of each observation."""
         # If we have predictions, then we have a thinned trace which does not
         # support extracting a log likelihood.
-        cached = [(var, var.logp_elemwise) for var in self.model.observed_RVs]
-        log_likelihood_dict = {}
+        if self.log_likelihood is True:
+            cached = [(var, var.logp_elemwise) for var in self.model.observed_RVs]
+        else:
+            cached = [
+                (var, var.logp_elemwise)
+                for var in self.model.observed_RVs
+                if var.name in self.log_likelihood
+            ]
+        log_likelihood_dict = self.pymc3.sampling._DefaultTrace(  # pylint: disable=protected-access
+            len(self.trace.chains)
+        )
```

Comment:
we should wrap this in a try except block, so that when pymc3 version is an old one, the error risen explains that pymc3 should be upgraded or arviz downgraded, otherwise users won't know what hit them.

There is no need to edit requirements file though, as pymc3 development is shown there.